ark/utils/segmentation_utils.py

```diff
@@ -127,3 +138,149 @@ def concatenate_csv(base_dir, csv_files, column_name="point", column_values=None
             combined_data = pd.concat((combined_data, temp_data), axis=0, ignore_index=True)
 
     combined_data.to_csv(os.path.join(base_dir, "combined_data.csv"), index=False)
+
+
+def visualize_watershed_transform(segmentation_labels_xr, channel_data_xr,
```

Comment:
This is an extremely long function that I think can be broken up into several smaller functions. Also, if some of the code here is duplicated from other files, you should consider importing and calling those modules to avoid repeated code.